File: 알고리즘/김연우/Kotlin/10828.kt
import java.io.BufferedReader
import java.io.InputStreamReader

lateinit var stack: IntArray
var ptr = -1

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val testCase = br.readLine()!!.toInt()

    stack = IntArray(testCase)


    for (i in 0 until testCase) {
        when (val input = br.readLine()) {
            "top" -> println(top())
            "pop" -> println(pop())
            "empty" -> println(isEmpty())
            "size" -> println(size())
            else -> push(input.split(" ")[1].toInt())
        }
    }
}

fun pop(): Int {
    return if (ptr == -1) -1 else stack[ptr--]
}

fun push(n: Int) {
    stack[++ptr] = n
}

fun isEmpty(): Int {
    return if (ptr == -1) 1 else 0
}

fun size(): Int {
    return ptr + 1
}

fun top(): Int {
    return if (ptr == -1) -1 else stack[ptr]
}

File: 알고리즘/정지관/javascript/10828.js
let fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");
let stack = [];
let answer = [];

for (let i = 1; i <= parseInt(input[0]); i++) {
  let arr = input[i].split(" ");
  switch (arr[0]) {
    case "push":
      stack.push(arr[1]);
      break;
    case "pop":
      if (stack.length === 0) answer.push(-1);
      else answer.push(stack.pop());
      break;
    case "size":
      answer.push(stack.length);
      break;
    case "empty":
      if (stack.length === 0) {
        answer.push(1);
      } else {
        answer.push(0);
      }
      break;
    case "top":
      if (stack.length === 0) answer.push(-1);
      else answer.push(stack[stack.length - 1]);
      break;
  }
}

console.log(answer.join("\n"));
